### DISCUSSION TITLE
I can't get extract subs and chapter in a BDRemux file

### DISCUSSION THREAD

**Discussion by papa-pitufo** [staxrip/staxrip#745] [General]
Hi everibody

I use BD folder or BDRemux to do my mkv file.

When I use BD Folder, **StaxRip show me a window** with audios, subs and Chapter and I can choose whatever you want.  Finally **StaxRip leave in my temp folder** audios, subs and Chapter that I choose.  Everything is OK

But when I choose a BDRemux (mkv file)
Directly jump to DGIndex
StaxRip doesn't show me any window like BD folder, it working extracting.  ????
Finally StaxRip leave in my temp folderwithout audios, subs and chapters.
But
I can choose the audios that I want in my work because StaxRip shows me in my work window 
I can choose the subs that I want in my work because StaxRip shows me in my work window (container options)
I don't insert the chapter file, because I don't have it, StaxRip doesn't extract it

Finally That I need to do is:

Working only video and audio
Use mkvextract to extract subs and chapter
USe mkvtoolnix to mount film with video, audio, subs and chapter.

I'm sure that I do anything bad, or I don't enable or enable anything wrong but I don't find it.

_Anyone help me?_

Show us a window with ficticius name to show the files of my temp folder

 
![image](https://user-images.githubusercontent.com/47916118/126033252-d8cfde72-d61c-4703-8fc9-c80476025620.png)

Note: E-AC3 audios are the final audios, the original audios are, AC3 and DTS MA, but both of them aren't inside my temp folder.

**Comment by Dendraspis** [comment]
You should check the options and set the right settings,especially for the languages and the demuxing ... and of course the chapters.

![image](https://user-images.githubusercontent.com/50659978/126077982-d0c57029-e2ab-49c8-ab65-73dab336d0d3.png)

**Reply by papa-pitufo** [reply]
Thanks but I enable, I think right.

Let's ee

![image](https://user-images.githubusercontent.com/47916118/126079650-92612ed7-9934-4aca-8e88-501073262a5f.png)

But the problem is only show dialog when I choose a BD folder, when I choose a mkv file doesn't show me.  Execute DGIndex directly

With BD folder, DGIndex is after staxrip demux audios, subs and chapter.

**Comment by JJKylee** [comment]
> But the problem is only show dialog when I choose a BD folder, when I choose a mkv file doesn't show me. Execute DGIndex directly
> 
> With BD folder, DGIndex is after staxrip demux audios, subs and chapter.

By `DGIndex`, you must be meaning `DGIndexNV`, because `DGIndex` cannot handle MKV files, aren't you?

So, that's because `DGIndexNV` has priority over `mkvextract` in handling MKV files according to your setting in _Settings_ > _Preprocessing_.

Put `mkvextract` **_over_** `DGIndexNV` in _Settings_ > _Preprocessing_ so that it can handle MKV files _before_ `DGIndexNV`.

See [this section on the `DGDecNV` wiki document](https://github.com/staxrip/staxrip/wiki/How-to-register-DGDecNV(DGSource-and-DGIndexNV)-in-StaxRip#-caveat-) for more info.

**Comment by papa-pitufo** [comment]
Hi @JJKylee , sorry but something I do wrong because don't run right.

I move mkvextract upper, above DGIndexNV (both DGIndexNV demux and no demux, following your steps in wiki document.)
I choose my mkv file and you're right show me a window where I can choose audios, subs, demux video and demux chapter.

Theorically works fine, then DGIndex work to make dgi file to my project (I see DGIndex, I don't see DGIndexNV)

I modify whatever I want, and do my work, but....
If I stop, suddenly shows errors in DGIndex, 3 differente errors, forever about DGIndex.

I fact, I stop it because the size of my work is smaller than thought.

Ok, I try to do it again, and now, shows the error **before** than I push start my work.

I change the order, and go down mkvextract below both DGIndex (Demux and no demux)

I start again, and now, everything works fine, without any erros, but, like I tell you at the begening of this post, I can't choose anything to demux.

I'm sure, that I am doing anything wrong, but I can't find it.  Sorry.

Could you help me?

Best regards

**Reply by JJKylee** [reply]
> (I see DGIndex, I don't see DGIndexNV)

`DGIndex` is _not_ supposed to handle **MKV** files. If this happens when you open an MKV file, your setup is wrong.

I can't reproduce your issues on my side.

And when you talk about errors, don't just _mention_ or _describe_ them. **_No logs, no analysis_**.

**Reply by papa-pitufo** [reply]
I try to reproduce, thanks in advantage

**Reply by papa-pitufo** [reply]
> > (I see DGIndex, I don't see DGIndexNV)
> 
> `DGIndex` is _not_ supposed to handle **MKV** files. If this happens when you open an MKV file, your setup is wrong.
> 
> I can't reproduce your issues on my side.
> 
> And when you talk about errors, don't just _mention_ or _describe_ them. **_No logs, no analysis_**.

Sorry @JJKylee but I try to reproduce the error to show the message and I don't get it.

Now, It start job right and when I stop, stops right.
No problem.

Very, very sorry, the other day I reproduced at least three times since the beginning and with 2 kind of files, because of this I write this post.

I don't know how to apologize for wasting your time. 

I'm really sorry. I am very ashamed

**Reply by JJKylee** [reply]
Maybe after you changed the settings, you forgot to restart StaxRip the other day.
Anyway, it's good that your issues are resolved now. 😊

**Comment by papa-pitufo** [comment]
Ok, I try to attach logs or analysis but I can't find it.
Where are these files?

Because in %StaxRip/Settings/Logfiles these files aren not.  Very strange, isn't it?

**Reply by JJKylee** [reply]
> If I stop, suddenly shows errors in DGIndex, 3 differente errors, forever about DGIndex.

By 'logs', I mean _the **error messages**_ of these errors.

In the StaxRip log folder, logs are written _only when the jobs are finished successfully_ in StaxRip. It's _not strange_ that unfinished jobs do not record logs there.

Another type of logs are recorded in the _project temp folder_. There you can find the log for unfinished jobs.

BTW, stopping indexers like `DGIndexNV` in the middle of working may produce unexpected results. Indexers need a certain amount of time for their jobs, and it increases proportionately with the input file size. I don't recommend stopping the indexing job.

**Comment by papa-pitufo** [comment]
Now, after the changes, where I move mkvextract, my preprocessor windows is

![image](https://user-images.githubusercontent.com/47916118/126361986-dfd5c8df-5414-46d7-a3da-b4d5f3cbb380.png)
![image](https://user-images.githubusercontent.com/47916118/126362187-35e6a167-8445-427e-ac75-f6603e66078a.png)
![image](https://user-images.githubusercontent.com/47916118/126362283-cb24c366-3726-4168-bdbd-4f148e2e22e6.png)
